### DISCUSSION TITLE
Converting a config file (module) to the Dagster way of doing

### DISCUSSION THREAD

**Discussion by ant1j** [dagster-io/dagster#17620] [Q&A] [answered]
Hello,
After a couple of weeks reading docs and tutorials, I am trying to convert one of my script to Dagster.
I have a bunch of functions linearly connected, making changes to a dataframe and then publish images and ppt decks.
I also have a python file (module) that I use as a config for all these functions. I have different configs in a `config` folder that I use for different runs - I only load 1 config module for a run (`from .config import the_right_config as config`).
This config module is a parameter for most of my functions (I know this is propbably not the "perfect" way to manage this, but this is how it is...)

I am stuck on integrating this config file in the Dagster way of things.

- I tend to consider that this config should probably be a `Resource ` since I can load the module and push it as a value for a `config` key in the resrouces in the `Definition`, and then use it as a parameter for my functions converted to `op` or `asset`.
- I follow the docs to make my config object as a `ResourceParam` for my functions. But this does not work: Dagster complains that it is not the right type, as I used `ResourceParam[ModuleType]`  as a type

```
dagster._core.errors.DagsterInvalidConfigDefinitionError: Error defining config. Original value passed: typing.Annotated[module, 'resource_param']. typing.Annotated[module, 'resource_param'] cannot be resolved.
This value can be a:
    - Field
    - Python primitive types that resolve to dagster config types
        - int, float, bool, str, list.
    - A dagster config type: Int, Float, Bool, Array, Optional, Selector, Shape, Permissive, Map
    - A bare python dictionary, which is wrapped in Field(Shape(...)). Any values
      in the dictionary get resolved by the same rules, recursively.
    - A python list with a single entry that can resolve to a type, e.g. [int]
```

Why do I use a module? because I am lazy and find that using attribute is easier than dict keys, and also because there are a lot of things in this file that I can easily access quickly from 1 object (again I guess this is not the right way to do but this is a starting point).

=> What would you recommend for me tu use?
=> Is there a way to use a module as a `Resource` for `op` or `asset`?

Thanks

**Comment by ant1j** [comment, marked as answer]
OK, one solution is not to name this resource `config` in the parameter of the function, as Dagster will expect the object to be a `Config` object and thus will try to validate it.
=> Is there a way to reamp the name to `config` even though it is a "reserved" word for Dagster, just like the `AssetIn` in the `ins` key of an `@asset` decorator?

**Reply by sryza** [reply]
> => Is there a way to reamp the name to config even though it is a "reserved" word for Dagster, just like the AssetIn in the ins key of an @asset decorator?

Alas this is currently not possible, and we're not likely to make this possible. However, I've filed an issue to track producing a less confusing error in this case: https://github.com/dagster-io/dagster/issues/17631.
